selene_sdk/samplers/multi_sampler.py

```diff
@@ -0,0 +1,324 @@
+        """
+        This method fetches a subset of the data from the sampler,
+        divided into batches. This method also allows the user to
+        specify what operating mode to run the sampler in when fetching
+        the data.
+
+        Parameters
+        ----------
+        batch_size : int
+            The size of the batches to divide the data into.
+        n_samples : int
+            The total number of samples to retrieve.
+        mode : str, optional
+            Default is None. The operating mode that the object should run in.
+            If None, will use the current mode `self.mode`.
+        """
+        if self._samplers[mode if mode else self.mode]:
+            return self._samplers[mode if mode else self.mode].get_data_and_targets(
+                batch_size, n_samples)
+        else:
+            self._set_batch_size(batch_size, mode=mode)
+            data_and_targets = []
+            targets_mat = []
+            count = batch_size
+            while count < n_samples:
+                data, tgts = self.sample(batch_size=batch_size, mode=mode)
+                data_and_targets.append((data, tgts))
+                targets_mat.append(tgts)
+                count += batch_size
+            remainder = batch_size - (count - n_samples)
+            data, tgts = self.sample(batch_size=remainder)
+            data_and_targets.append((data, tgts))
+            targets_mat.append(tgts)
+            targets_mat = np.vstack(targets_mat)
+            return data_and_targets, targets_mat
+
+    def get_validation_set(self, batch_size, n_samples=None):
+        """
+        This method returns a subset of validation data from the
+        sampler, divided into batches.
+
+        Parameters
+        ----------
+        batch_size : int
+            The size of the batches to divide the data into.
+        n_samples : int, optional
+            Default is None. The total number of validation examples to
+            retrieve. Handling for `n_samples=None` should be done by
+            all classes that subclass `selene_sdk.samplers.Sampler`.
+
+        """
+        return self.get_data_and_targets(
+            batch_size, n_samples, mode="validate")
+
+    def get_test_set(self, batch_size, n_samples=None):
+        """
+        This method returns a subset of testing data from the
+        sampler, divided into batches.
+
+        Parameters
+        ----------
+        batch_size : int
+            The size of the batches to divide the data into.
+        n_samples : int or None, optional
+            Default is `None`. The total number of validation examples
+            to retrieve. If `None`, 640000 examples are retrieved.
+
+        Returns
+        -------
+        sequences_and_targets, targets_matrix : \
+        tuple(list(tuple(numpy.ndarray, numpy.ndarray)), numpy.ndarray)
+            Tuple containing the list of sequence-target pairs, as well
```

Comment:
I might clarify this documentation - I know I must have written/specified this originally, but just to double check, this is basically `sequence_and_targets` is the sequence-target in batches, and `targets_matrix` is just not batched version right? (without sequences) for ease of iteration or something

Reply:
yes `sequence_and_targets`  is a list of (sequence, targets) tuples and `targets_matrix` is a combined array of targets. I don't remember why but it has to be for the convenience of doing something downstream